src/io/iter_image_recordio_2.cc

```diff
@@ -285,9 +285,9 @@ inline bool ImageRecordIOParser2<DType>::ParseNext(DataBatch *out) {
     shape_vec.push_back(param_.label_width);
     TShape label_shape(shape_vec.begin(), shape_vec.end());
 
-    out->data.at(0) = NDArray(data_shape, Context::CPUPinned(0), false,
+    out->data.at(0) = NDArray(data_shape, Context::CPU(0), false,
```

Comment:
For my own understanding, why do we change to CPU from CPUPinned in this PR? Is it going to cause issue if we stick to CPUPinned when conduct distributed training with Horovod?

Reply:
In my benchmark, it doesn't affect performance. You may recall we can't use `horovod.local_rank()` to determine the GPU id because `mxnet` cannot be compile-time dependent on `horovod`. But then if you set all GPUs from 0-7 to use `CPUPinned(0)` you will get 8 processes starting the CUDA driver on GPU 0, which wastes memory and you won't be able to get the largest batch size.

Reply:
@ctcyang what did you benchmark? Did you also run kvstore('local') and kvstore('nccl') to verify the perf impact?